File: app/src/main/java/com/zhiyicx/thinksnsplus/modules/wallet/integration/detail/recharge_withdrawal/IntegrationRWDetailContainerFragment.java
package com.zhiyicx.thinksnsplus.modules.wallet.integration.detail.recharge_withdrawal;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.View;

import com.zhiyicx.baseproject.base.TSViewPagerFragment;
import com.zhiyicx.thinksnsplus.R;
import com.zhiyicx.thinksnsplus.modules.wallet.integration.detail.IntegrationDetailListFragment;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import static com.zhiyicx.thinksnsplus.modules.wallet.integration.detail.IntegrationDetailListFragment.CHOOSE_TYPE_CASH;
import static com.zhiyicx.thinksnsplus.modules.wallet.integration.detail.IntegrationDetailListFragment.CHOOSE_TYPE_RECHARGE;

/**
 * @Describe
 * @Author Jungle68
 * @Date 2018/1/24
 * @Contact master.jungle68@gmail.com
 */

public class IntegrationRWDetailContainerFragment extends TSViewPagerFragment {
    public static final String BUNDLE_DEFAULT_POSITION = "default_postion";

    public static final int POSITION_RECHARGE_RECORD = 0;
    public static final int POSITION_WITHDRAWASL_RECORD = 1;

    private int mDefaultPosition;

    public static IntegrationRWDetailContainerFragment newInstance(int default_position) {

        Bundle args = new Bundle();
        IntegrationRWDetailContainerFragment fragment = new IntegrationRWDetailContainerFragment();
        args.putInt(BUNDLE_DEFAULT_POSITION, default_position);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    protected boolean setUseSatusbar() {
        return true;
    }

    @Override
    protected boolean setUseStatusView() {
        return true;
    }


    @Override
    protected List<String> initTitles() {
        return Arrays.asList(getResources().getStringArray(R.array.integration_recharge_withdrawal_type));
    }

    @Override
    protected List<Fragment> initFragments() {
        if (mFragmentList == null) {
            mFragmentList = new ArrayList();
            mFragmentList.add(IntegrationDetailListFragment.newInstance(CHOOSE_TYPE_RECHARGE));
            mFragmentList.add(IntegrationDetailListFragment.newInstance(CHOOSE_TYPE_CASH));
        }
        return mFragmentList;
    }


    @Override
    protected void initViewPager(View rootView) {
        super.initViewPager(rootView);
        mTsvToolbar.setDefaultTabLeftMargin(getResources().getDimensionPixelOffset(R.dimen.spacing_big_large));
        mTsvToolbar.setDefaultTabRightMargin(getResources().getDimensionPixelOffset(R.dimen.spacing_big_large));
        mDefaultPosition = getArguments().getInt(BUNDLE_DEFAULT_POSITION);
        if (mDefaultPosition == POSITION_WITHDRAWASL_RECORD) {
            mVpFragment.setCurrentItem(mDefaultPosition);
        }

    }

    @Override
    protected void initData() {

    }
}

File: app/src/main/java/com/zhiyicx/thinksnsplus/modules/wallet/integration/detail/recharge_withdrawal/IntegrationRWDetailContainerFragment.kt
package com.zhiyicx.thinksnsplus.modules.wallet.integration.detail.recharge_withdrawal

import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.View

import com.zhiyicx.baseproject.base.TSViewPagerFragment
import com.zhiyicx.thinksnsplus.R
import com.zhiyicx.thinksnsplus.modules.wallet.integration.detail.IntegrationDetailListFragment
import com.zhiyicx.thinksnsplus.modules.wallet.integration.detail.IntegrationDetailPresenter

import java.util.ArrayList
import java.util.Arrays

/**
 * @Describe
 * @Author Jungle68
 * @Date 2018/1/24
 * @Contact master.jungle68@gmail.com
 */

class IntegrationRWDetailContainerFragment : TSViewPagerFragment<IntegrationDetailPresenter>() {

    override fun setUseSatusbar() = true

    override fun setUseStatusView() = true

    override fun initTitles() = Arrays.asList(*resources.getStringArray(R.array.integration_recharge_withdrawal_type))

    override fun initFragments(): List<Fragment> {
        if (mFragmentList == null) {
            mFragmentList = ArrayList()
            mFragmentList.add(IntegrationDetailListFragment.newInstance(IntegrationDetailListFragment.CHOOSE_TYPE_RECHARGE))
            mFragmentList.add(IntegrationDetailListFragment.newInstance(IntegrationDetailListFragment.CHOOSE_TYPE_CASH))
        }
        return mFragmentList
    }

    override fun initViewPager(rootView: View) {
        super.initViewPager(rootView)
        mTsvToolbar.setDefaultTabLeftMargin(resources.getDimensionPixelOffset(R.dimen.spacing_big_large))
        mTsvToolbar.setDefaultTabRightMargin(resources.getDimensionPixelOffset(R.dimen.spacing_big_large))
        val defaultPosition = arguments.getInt(BUNDLE_DEFAULT_POSITION)
        if (defaultPosition == POSITION_WITHDRAWASL_RECORD) {
            mVpFragment.currentItem = defaultPosition
        }

    }

    override fun initData(){}

    companion object {
        const val BUNDLE_DEFAULT_POSITION = "default_postion"

        const val POSITION_RECHARGE_RECORD = 0
        const val POSITION_WITHDRAWASL_RECORD = 1

        fun newInstance(defaultPosition: Int): IntegrationRWDetailContainerFragment {

            val args = Bundle()
            val fragment = IntegrationRWDetailContainerFragment()
            args.putInt(BUNDLE_DEFAULT_POSITION, defaultPosition)
            fragment.arguments = args
            return fragment
        }
    }
}
